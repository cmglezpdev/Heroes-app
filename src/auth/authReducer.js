import { types } from "../types/types";

// const state = {
//     name: 'Carlos',
//     logged: true
// }


export const authReducer = ( state = {}, action ) => {

    switch( action.type ) {
        case types.login:
            return {
                ...action.payload,
                logged: true
            }

        case types.logout:
            return {
                logged: false
            }

        default:
            return state;
    }
}