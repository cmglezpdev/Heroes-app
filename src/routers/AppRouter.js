import React from 'react'
import { Routes, Route, BrowserRouter } from 'react-router-dom';
import { DcScreem } from '../components/dc/DcScreem';
import { LoginScreen } from '../components/login/LoginScreen';
import { MarvelScreen } from '../components/marvel/MarvelScreen';
import { SearchScreen } from '../components/search/SearchScreen';
import { Navbar } from '../components/ui/NavBar';

export const AppRouter = () => {
  return (
    <BrowserRouter>
        <Navbar />
        
        <Routes>
            <Route path="/" element={<MarvelScreen />} />
            <Route path="/marvel" element={<MarvelScreen />} />
            <Route path="/dc" element={<DcScreem />} />
            <Route path="/search" element={<SearchScreen />} />
            <Route path="/login" element={<LoginScreen />} />

        </Routes>
    </BrowserRouter>
  )
}
