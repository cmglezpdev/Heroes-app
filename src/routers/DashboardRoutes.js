import { Routes, Route } from 'react-router-dom';
import { DcScreem } from '../components/dc/DcScreem';
import { MarvelScreen } from '../components/marvel/MarvelScreen';
import { SearchScreen } from '../components/search/SearchScreen';
import { Navbar } from '../components/ui/NavBar';
import { HeroScreen } from '../components/hero/HeroScreen';

export const DashboardRoutes = () => {
  return (
    <>
        <Navbar />
        <div className='container'>
          <Routes>
              <Route path="/" element={<MarvelScreen />} />
              <Route path="marvel" element={<MarvelScreen />} />
              <Route path="dc" element={<DcScreem />} />
              <Route path="search" element={<SearchScreen />} />
              <Route path="hero" element={<HeroScreen />} />
          </Routes>
        </div>
    </>
  )
}
