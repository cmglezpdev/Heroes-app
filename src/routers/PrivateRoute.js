import { useContext } from "react"
import { AuthContext } from "../auth/authContext";
import { Navigate } from "react-router-dom";

export const PrivateRoute = ({ children }) => {

    const { user } = useContext(AuthContext);


    return user.logged ?
        children :
        (<Navigate to="/login" />)
}
