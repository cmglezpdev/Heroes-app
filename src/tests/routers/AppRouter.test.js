import { prettyDOM, render, screen } from "@testing-library/react";
import { AppRouter } from "../../routers/AppRouter";
import { AuthContext } from '../../auth/authContext';


describe('tests about AppRouter', () => {

    test('should show login page if not be logged', () => {                 
        const contextValue = {
            user: {
                logged: false
            }
        }
        render(
            <AuthContext.Provider value={contextValue}>
                <AppRouter />
            </AuthContext.Provider>
        );
        
        const h1 = screen.getByRole('heading');
        expect( h1.textContent ).toBe('Login');
    });

    test('should show marvel component if be logged', () => {
        const contextValue = {
            user: {
                name: 'Carlos Manuel',
                logged: true
            }
        }                 
        const { container } = render(
            <AuthContext.Provider value={contextValue}>
                <AppRouter />
            </AuthContext.Provider>
        );
        expect( container ).toMatchSnapshot();
        const nav = screen.getByRole('navigation');
        expect(nav).toBeDefined();


        // console.log(prettyDOM(container));
    });

})