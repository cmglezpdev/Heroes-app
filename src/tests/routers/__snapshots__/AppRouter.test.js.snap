// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests about AppRouter should show marvel component if be logged 1`] = `
<div>
  <nav
    class="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      class="navbar-brand"
      href="/"
    >
      Asociaciones
    </a>
    <div
      class="navbar-collapse"
    >
      <div
        class="navbar-nav"
      >
        <a
          class="nav-item nav-link "
          href="/marvel"
        >
          Marvel
        </a>
        <a
          class="nav-item nav-link "
          href="/dc"
        >
          DC
        </a>
        <a
          class="nav-item nav-link "
          href="/search"
        >
          Search
        </a>
      </div>
    </div>
    <div
      class="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
    >
      <ul
        class="navbar-nav ml-auto"
      >
        <span
          class="nav-link nav-item text-info"
        >
          Carlos Manuel
        </span>
        <button
          class="nav-item nav-link btn"
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>
  <div
    class="container"
  >
    <div>
      <h1>
        Marvel Comics
      </h1>
      <div
        class="row rows-cols-1 row-cols-md-3 g-3 animate__animated animate__fadeIn"
      >
        <div
          class="col animate__animated animate__fadeIn"
        >
          <div
            class="card"
          >
            <div
              class="row no-gutters"
            >
              <div
                class="col-md-4"
              >
                <img
                  alt="Spider Man"
                  class="card-img"
                  src="/assets/marvel-spider.jpg"
                />
              </div>
              <div
                class="col-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-title"
                  >
                    Spider Man
                  </h5>
                  <p
                    class="card-text"
                  >
                    Peter Parker
                  </p>
                  <a
                    href="/hero/marvel-spider"
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col animate__animated animate__fadeIn"
        >
          <div
            class="card"
          >
            <div
              class="row no-gutters"
            >
              <div
                class="col-md-4"
              >
                <img
                  alt="Captain America"
                  class="card-img"
                  src="/assets/marvel-captain.jpg"
                />
              </div>
              <div
                class="col-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-title"
                  >
                    Captain America
                  </h5>
                  <p
                    class="card-text"
                  >
                    Steve Rogers
                  </p>
                  <a
                    href="/hero/marvel-captain"
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col animate__animated animate__fadeIn"
        >
          <div
            class="card"
          >
            <div
              class="row no-gutters"
            >
              <div
                class="col-md-4"
              >
                <img
                  alt="Iron Man"
                  class="card-img"
                  src="/assets/marvel-iron.jpg"
                />
              </div>
              <div
                class="col-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-title"
                  >
                    Iron Man
                  </h5>
                  <p
                    class="card-text"
                  >
                    Tony Stark
                  </p>
                  <a
                    href="/hero/marvel-iron"
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col animate__animated animate__fadeIn"
        >
          <div
            class="card"
          >
            <div
              class="row no-gutters"
            >
              <div
                class="col-md-4"
              >
                <img
                  alt="Thor"
                  class="card-img"
                  src="/assets/marvel-thor.jpg"
                />
              </div>
              <div
                class="col-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-title"
                  >
                    Thor
                  </h5>
                  <p
                    class="card-text"
                  >
                    Thor Odinson
                  </p>
                  <a
                    href="/hero/marvel-thor"
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col animate__animated animate__fadeIn"
        >
          <div
            class="card"
          >
            <div
              class="row no-gutters"
            >
              <div
                class="col-md-4"
              >
                <img
                  alt="Hulk"
                  class="card-img"
                  src="/assets/marvel-hulk.jpg"
                />
              </div>
              <div
                class="col-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-title"
                  >
                    Hulk
                  </h5>
                  <p
                    class="card-text"
                  >
                    Bruce Banner
                  </p>
                  <a
                    href="/hero/marvel-hulk"
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col animate__animated animate__fadeIn"
        >
          <div
            class="card"
          >
            <div
              class="row no-gutters"
            >
              <div
                class="col-md-4"
              >
                <img
                  alt="Wolverine"
                  class="card-img"
                  src="/assets/marvel-wolverine.jpg"
                />
              </div>
              <div
                class="col-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-title"
                  >
                    Wolverine
                  </h5>
                  <p
                    class="card-text"
                  >
                    James Howlett
                  </p>
                  <a
                    href="/hero/marvel-wolverine"
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col animate__animated animate__fadeIn"
        >
          <div
            class="card"
          >
            <div
              class="row no-gutters"
            >
              <div
                class="col-md-4"
              >
                <img
                  alt="Daredevil"
                  class="card-img"
                  src="/assets/marvel-daredevil.jpg"
                />
              </div>
              <div
                class="col-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-title"
                  >
                    Daredevil
                  </h5>
                  <p
                    class="card-text"
                  >
                    Matthew Michael Murdock
                  </p>
                  <a
                    href="/hero/marvel-daredevil"
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col animate__animated animate__fadeIn"
        >
          <div
            class="card"
          >
            <div
              class="row no-gutters"
            >
              <div
                class="col-md-4"
              >
                <img
                  alt="Hawkeye"
                  class="card-img"
                  src="/assets/marvel-hawkeye.jpg"
                />
              </div>
              <div
                class="col-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-title"
                  >
                    Hawkeye
                  </h5>
                  <p
                    class="card-text"
                  >
                    Clinton Francis Barton
                  </p>
                  <a
                    href="/hero/marvel-hawkeye"
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col animate__animated animate__fadeIn"
        >
          <div
            class="card"
          >
            <div
              class="row no-gutters"
            >
              <div
                class="col-md-4"
              >
                <img
                  alt="Cyclops"
                  class="card-img"
                  src="/assets/marvel-cyclops.jpg"
                />
              </div>
              <div
                class="col-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-title"
                  >
                    Cyclops
                  </h5>
                  <p
                    class="card-text"
                  >
                    Scott Summers
                  </p>
                  <a
                    href="/hero/marvel-cyclops"
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col animate__animated animate__fadeIn"
        >
          <div
            class="card"
          >
            <div
              class="row no-gutters"
            >
              <div
                class="col-md-4"
              >
                <img
                  alt="Silver Surfer"
                  class="card-img"
                  src="/assets/marvel-silver.jpg"
                />
              </div>
              <div
                class="col-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-title"
                  >
                    Silver Surfer
                  </h5>
                  <p
                    class="card-text"
                  >
                    Norrin Radd
                  </p>
                  <a
                    href="/hero/marvel-silver"
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
